src/components/LocationSelector/LocationSelector.js

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { LoadScriptNext } from '@react-google-maps/api';
+import {
+    Modal,
+    Map,
+    FieldWrapper,
+    LocationAutocomplete,
+    Button,
+    LocationCard,
+} from '@textkernel/oneui';
+
+const DEFAULT_LOCATION_CIRCLE_RADIUS = 1;
+
+function LocationSelector(props) {
+    const {
+        minRadius,
+        maxRadius,
+        selectedLocations,
+        addLocation,
+        changeLocation,
+        removeLocation,
+        doneLabel,
+    } = props;
+
+    const [isOpen, setIsOpen] = React.useState(false);
+
+    /**
+     * Fetch additional information for the selected place and
+     * add it along with passed location object to the selectedLocations array
+     */
+    function onAddLocation(location) {
+        const placeInfo = fetchPlaceInfo(location.id);
+        const locationToAdd = {
+            ...location,
+            center: placeInfo.coordinates,
+            radius: DEFAULT_LOCATION_CIRCLE_RADIUS,
+            structuredFormatting: placeInfo.structuredFormatting,
+        };
+        addLocation(locationToAdd);
+    }
+
+    return (
+        <>
+            <FieldWrapper />
+            <Modal isOpen={isOpen}>
+                <LoadScriptNext>
+                    <LocationAutocomplete />
+                    <Button context="brand">{doneLabel}</Button>
+                    {selectedLocations.map(location => (
+                        <LocationCard
+                            locationId={location.id}
+                            locationTitle={location.description}
+                            sliderLabel={location.radius}
+                            minRadius={minRadius}
+                            maxRadius={maxRadius}
+                            // lots of other props
+                        />
+                    ))}
+                    <Map />
+                </LoadScriptNext>
+            </Modal>
+        </>
+    );
+}
+
+LocationSelector.displayName = 'LocationSelector';
+
+LocationSelector.propTypes = {
+    /** Google api key */
+    apiKey: PropTypes.string.isRequired,
+    /** stores an array of selected location objects */
+    selectedLocations: PropTypes.arrayOf(
```

Comment:
will it work ok without this array? Maybe should be required if this to be implemented as a controlled component